stock_picking_warn_message/models/stock_picking.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    picking_warn = fields.Text(compute="_compute_picking_warn")
```

Comment:
```suggestion
from odoo import api, fields, models
from odoo.addons.base.models.res_partner import WARNING_HELP, WARNING_MESSAGE


class StockPicking(models.Model):
    _inherit = "stock.picking"

    picking_warn = fields.Selection(selection=WARNING_MESSAGE, compute="_compute_picking_warn", help=WARNING_HELP)
```
I believe this field should have been declared as a Selection, following the related field as done here: https://github.com/odoo/odoo/blob/17.0/addons/stock/models/res_partner.py#L20.
I guess we can use the migration to improve this, that way the functionality is more consistent.
Then, we should also transition the data with a pre-migration script.